core/services/offchainreporting/run_saver.go

```diff
@@ -64,8 +74,14 @@ func (r *RunResultSaver) Close() error {
 		select {
 		case rr := <-r.runResults:
 			logger.Debugw("RunSaver: saving job run before exiting", "run", rr.Run, "task results", rr.TaskRunResults)
-			if _, err := r.pipelineRunner.InsertFinishedRun(context.Background(), rr.Run, rr.TaskRunResults, false); err != nil {
-				logger.Errorw(fmt.Sprintf("error inserting finished results for job %v", r.jobID), "err", err)
+			ctx, cancel := postgres.DefaultQueryCtx()
+			defer cancel()
+			err := postgres.GormTransaction(ctx, r.db, func(tx *gorm.DB) error {
```

Comment:
What is the point of this transaction, it only has one DB operation in it?

Reply:
it doesn't, InsertFinishedRun does lots of stuff inside

Reply:
Ah gotcha. I would have expected InsertFinishedRun to do its own transaction in that case